Classes/Security/FilePermissionsAspect.php

```diff
@@ -57,12 +71,11 @@ protected function respectStorage(Matcher $matcher) {
 	}
 
 	/**
-	 * @param Matcher $matcher
+	 * @param Query $query
+	 * @param ConstraintInterface|NULL $constraints
 	 * @return void
 	 */
-	protected function respectFileMounts(Matcher $matcher) {
-		$matcher->setLogicalSeparatorForLike(Matcher::LOGICAL_OR);
-
+	protected function respectFileMounts(Query &$query, &$constraints) {
```

Comment:
I am not sure why it seems necessary to use "&" for $query and $constraints here. AFAIK objects should automatically be passed by reference in PHP. But without those "&" it will not work.